cmake/VcMacros.cmake

```diff
@@ -495,7 +495,7 @@ macro(vc_compile_for_all_implementations _srcs _src)
       # which case AVX2 implies the availability of FMA and BMI2
       #_vc_compile_one_implementation(${_srcs} AVX2  "-mavx2")
       #_vc_compile_one_implementation(${_srcs} AVX2+BMI2 "-mavx2 -mbmi2")
-      _vc_compile_one_implementation(${_srcs} AVX2+FMA+BMI2 "-xCORE-AVX2" "-mavx2 -mfma -mbmi2" "/arch:AVX2")
+      _vc_compile_one_implementation(${_srcs} AVX2+FMA+BMI2 "-xAVX2" "-mavx2 -mfma -mbmi2" "/arch:AVX2")
```

Comment:
Yes. And funnily it works. Because the unit tests did not work with `-xAVX2`, I needed `-mavx2`. Should we change to `-mavx2` here as well?

Reply:
Yes, I think that anything that could be run on AMD should fallback to `-mavx2` so that the compiler doesn't add special code based on cpuid checks (for example, if may just fallback to scalar if the AMD CPU is not recognized, or even [intentionally](https://www.agner.org/optimize/blog/read.php?i=49)).